test/test_files/exceptions/copy/wrong_header.test

```diff
@@ -3,6 +3,78 @@
 
 --
 
+-CASE CSVHeaderMismatch
+-STATEMENT CREATE NODE TABLE person (ID INT64, fName STRING, PRIMARY KEY (ID))
+---- ok
+-STATEMENT COPY person FROM "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/vPerson.csv" (HEADER=true)
+---- ok
+-STATEMENT CREATE NODE TABLE person1 (ID STRING, fName STRING, PRIMARY KEY (ID))
+---- ok
+-STATEMENT CREATE NODE TABLE person2 (ID STRING, fName INT64, PRIMARY KEY (ID));
+---- ok
+-STATEMENT CREATE NODE TABLE person3 (ID STRING, PRIMARY KEY (ID));
+---- ok
+-STATEMENT COPY person1 FROM "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/vPerson.csv" (HEADER=true)
+---- ok
+-STATEMENT MATCH (a:person1) RETURN a.ID STARTS WITH '1';
+---- 3
+False
+False
+True
+-STATEMENT COPY person2 FROM "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/vPerson.csv" (HEADER=true)
+---- error
+Conversion exception: Cast failed. Guodong is not in INT64 range.
+-STATEMENT COPY person3 FROM "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/vPerson.csv" (HEADER=true)
+---- error
+Binder exception: Number of columns mismatch. Detect 2 but expect 1.
+-STATEMENT COPY person FROM
+       ["${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/vPerson.csv",
+        "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/vPersonMissingColumn.csv"] (HEADER=true)
+---- error
+Binder exception: Number of columns mismatch. Detect 1 but expect 2.
+-STATEMENT CREATE REL TABLE knows (FROM person TO person, prop1 INTERVAL);
+---- ok
+-STATEMENT COPY knows FROM "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/eKnowsWrongColumnName.csv" (HEADER=true)
+---- error
+Binder exception: Number of columns mismatch. Detect 4 but expect 3.
+-STATEMENT COPY knows FROM "${KUZU_ROOT_DIRECTORY}/dataset/copy-fault-tests/wrong-header/eKnowsMissingColumn.csv" (HEADER=true)
+---- error
+Conversion exception: Error occurred during parsing interval. Field name is missing.
+
+-CASE ParquetHeaderMismatch
+-STATEMENT CREATE NODE TABLE User(name STRING, age INT64, PRIMARY KEY (name));
+---- ok
+-STATEMENT CREATE NODE TABLE User1(name INT64, age INT64, PRIMARY KEY (name));
+---- ok
+-STATEMENT CREATE NODE TABLE User2(name INT64, age INT64, dummy INT32[], PRIMARY KEY (name));
+---- ok
+-STATEMENT CREATE NODE TABLE User3(name STRING, age INT16[], PRIMARY KEY (name));
+---- ok
+-STATEMENT COPY User FROM "${KUZU_ROOT_DIRECTORY}/dataset/demo-db/parquet/user.parquet";
+---- ok
+-STATEMENT COPY User1 FROM "${KUZU_ROOT_DIRECTORY}/dataset/demo-db/parquet/user.parquet";
+---- error
+Binder exception: Column name type mismatch. Expect INT64 but get STRING.
```

Comment:
Can you add a test case where the parquet file contains one  date column, and you want to import it to a int32 column.